File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryEventLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryEventLabelsTest {
  @Test
  fun `maps known delivery event types to readable labels`() {
    assertEquals("Saiu para entrega", deliveryEventTypeLabel("SAIU_PARA_ENTREGA"))
    assertEquals("Rota recalculada", deliveryEventTypeLabel("ROTA_RECALCULADA"))
    assertEquals("Notificacao enviada", deliveryEventTypeLabel("NOTIFICACAO_ENVIADA"))
  }

  @Test
  fun `formats unknown delivery event types without exposing raw underscores`() {
    assertEquals("Evento novo", deliveryEventTypeLabel("EVENTO_NOVO"))
  }

  @Test
  fun `uses a safe fallback for blank delivery event types`() {
    assertEquals("Evento", deliveryEventTypeLabel("   "))
  }
}

File: apps/motoboy-pwa/src/deliveryEventLabels.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { deliveryEventTypeLabel } from "./deliveryEventLabels";

test("maps known delivery event types to readable labels", () => {
  assert.equal(deliveryEventTypeLabel("SAIU_PARA_ENTREGA"), "Saiu para entrega");
  assert.equal(deliveryEventTypeLabel("ROTA_RECALCULADA"), "Rota recalculada");
  assert.equal(deliveryEventTypeLabel("NOTIFICACAO_ENVIADA"), "Notificacao enviada");
});

test("formats unknown delivery event types without exposing raw underscores", () => {
  assert.equal(deliveryEventTypeLabel("EVENTO_NOVO"), "Evento novo");
});

test("uses a safe fallback for blank delivery event types", () => {
  assert.equal(deliveryEventTypeLabel("   "), "Evento");
});
